### PR TITLE
Add branch naming convention

### PR DESCRIPTION
**Pull request by CristinaSescu**
Because:
- naming a git branch will itself need a standard convention to be followed in a project to keep it organized.

I found these rules which seem to me like a good start to naming branches.
https://dev.to/couchcamote/git-branching-name-convention-cch

This rules should resolve the ticket:
#7

### CODE REVIEW

**Comment by Cozonacci** [comment]
Check the view here and adjust for consistency and readability.
https://github.com/CiprianBucur/Java-project/tree/feature/add-branch-naming-convention

File: README.md [hunk 1]
```diff
@@ -45,3 +45,50 @@ Body:
 # Useful link for GitHub learning
 1. https://learngitbranching.js.org/
 
+# Git Branch Naming Convention
+
+# Temporary Branches -> As the name implies, these are 
+# disposable branches that can be created and deleted by need 
+# of the developer or deployer.
+ 
+1. Feature Any code changes for a new module or use case should be done
```
Comment (VoicuGeo) [inline README.md:54]: will be good to use : ? 
example: 
Feature: Any code changes.... 
or
**Feature** : Any code...

File: README.md [hunk 2]
```diff
@@ -45,3 +45,50 @@ Body:
...
+   on a feature branch. This branch is created based on the current development
+   branch. When all changes are Done, a Pull Request/Merge Request is needed 
+   to put all of these to the development branch.
+
+      Examples:
+      feature/integrate-swagger
+      feature/support-dark-theme
+
+2. Bug Fix If the code changes made from the feature branch were rejected after
```
Comment (VoicuGeo) [inline README.md:63]: the same here

File: README.md [hunk 3]
```diff
@@ -45,3 +45,50 @@ Body:
...
+   a release, sprint or demo, any necessary fixes after that should be done on 
+   the bugfix branch.
+
+      Examples:
+      bugfix/more-gray-shades
+      bugfix/gray-on-blur-fix
+      
+3. Hot Fix If there is a need to fix a blocker, do a temporary patch, apply a
```
Comment (VoicuGeo) [inline README.md:71]: same

File: README.md [hunk 4]
```diff
@@ -45,3 +45,50 @@ Body:
...
+   critical framework or configuration change that should be handled immediately,
+   it should be created as a Hotfix. It does not follow the scheduled integration 
+   of code and could be merged directly to the production branch, then on the
+   development branch later.
+
+      Examples:
+      hotfix/disable-endpoint-zero-day-exploit
+      hotfix/increase-scaling-threshold
+      
+4. Merging A temporary branch for resolving merge conflicts, usually between
```
Comment (VoicuGeo) [inline README.md:81]: same

**Pull request merged**